Add integration testing support for OBA servers and GTFS bundles

Abo-Omar-74 opened the pull request:
- Introduce integration tests for GTFS bundle downloads and OBA API availability
- Add helper function to load server configurations for integration tests
- Provide integration configuration template and setup instructions in README

aaronbrethorst reviewed (commented):
nice work! please rebase against main once I merge your previous PR and then take care of the comments I and Copilot made. can't wait to get this in!

Abo-Omar-74 commented:
I have made the requested changes:
- Rebased with the latest main
- Removed the comment from `integration_servers.json.template`

Regarding Copilot suggestions:  
I will open another PR to introduce context usage in the function that handles the cron job across the application.

aaronbrethorst commented:
@Abo-Omar-74 ready for rebase

Abo-Omar-74 commented:
@aaronbrethorst done rebasing - it's ready to merge!
